auth/client/iam/client.go

```diff
@@ -90,6 +90,23 @@ func (hb HTTPClient) OAuthAuthorizationServerMetadata(ctx context.Context, webDI
 	return &metadata, nil
 }
 
+// ClientMetadata retrieves the client metadata from the client metadata endpoint given in the authorization request.
+// We use the AuthorizationServerMetadata struct since it overlaps greatly with the client metadata.
+func (hb HTTPClient) ClientMetadata(ctx context.Context, endpoint string) (*oauth.AuthorizationServerMetadata, error) {
```

Comment:
its slightly confusing that HTTPClient implements 3 out 4 methods defined in services/oauth#Holder interface but is not intended as an implementation of the interface.